From a5b1e5feee96b753b080c640f6837c13f1dd94d1 Mon Sep 17 00:00:00 2001
From: Suraj P Kizhakkethil <quic_surapk@quicinc.com>
Date: Thu, 23 Jan 2025 09:35:31 +0530
Subject: [PATCH] QSDK:CP: mac80211.sh: Add support for DPP

Invoke dpp action script when dpp is enabled for any interface

Change-Id: I42e14aed5266946ee1b2790c1d7a4d166a31ee60
Signed-off-by: Suraj P Kizhakkethil <quic_surapk@quicinc.com>
(cherry picked from commit 3520d56a52cc47fb990cfa8d5a2c42c13aea1a72)
---
 .../wifi-scripts/files/lib/netifd/wireless/mac80211.sh     | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/package/network/config/wifi-scripts/files/lib/netifd/wireless/mac80211.sh b/package/network/config/wifi-scripts/files/lib/netifd/wireless/mac80211.sh
index 884cd9013074..02dba75a1a3b 100755
--- a/package/network/config/wifi-scripts/files/lib/netifd/wireless/mac80211.sh
+++ b/package/network/config/wifi-scripts/files/lib/netifd/wireless/mac80211.sh
@@ -80,6 +80,9 @@ dtim_period=
 max_listen_int=
 he_6ghz_reg_pwr_type=
 
+#dpp
+dpp_ifaces=
+
 wdev_tool() {
 	ucode /usr/share/hostap/wdev.uc "$@"
 }
@@ -1144,6 +1147,8 @@ mac80211_prepare_vif() {
 
 			mac80211_hostapd_setup_bss "$phy" "$ifname" "$macaddr" "$type" || return
 
+			[ -n "$dpp" ] && append dpp_ifaces $ifname
+
 			[ -n "$hostapd_ctrl" ] || {
 				ap_ifname="${ifname}"
 				hostapd_ctrl="${hostapd_ctrl:-/var/run/hostapd/$ifname}"
@@ -1848,6 +1853,8 @@ drv_mac80211_setup() {
 		. /lib/performance.sh
 	}
 
+	hostapd_dpp_action "$dpp_ifaces"
+
 	for_each_interface "ap mesh" mac80211_set_fq_limit
 	wireless_set_up
 }
-- 
2.34.1

